eos-tech-support/eos-write-live-image

```diff
@@ -45,7 +45,8 @@ LABEL=
 DEBUG=
 
 usage() {
-    local SELF="$(basename "$0")"
+    local SELF
+    SELF=$(basename "$0")
```

Comment:
Hmm, that's an interesting one. I wonder what was wrong with the original.

Reply:
https://gist.github.com/wjt/2bcb4a9705db683c8d8292e3cc4ac9d3

```
  local x=$(echo oh no; false)
        ^-- SC2155: Declare and assign separately to avoid masking return values.
```

Reply:
Also the outer quotes are redundant, I think

Reply:
Interesting. Ah, this is (as the warning says) when you're mixing declarations (`local`, `export`, etc) with assignment. Good to know!
```
$ x=$(false)
$ echo $?
1
$ export x=$(false)
$ echo $?
0
$ f() { local x=$(false); echo $?; }
$ f
0
```
I always leave the quotes out when doing variable assignment with command substitution, but only because I've experienced it to be unnecessary in bash and dash. I'm not sure why that's the case, though.

Reply:
`man dash` does not have anything much to say on this. `man bash` says under `PARAMETERS` (emphasis mine):

> A variable may be assigned to by a statement of the form
>
> ```bash
> name=[value]
> ```
>
> If  value is not given, the variable is assigned the null string.  All values undergo tilde expansion, parameter and variable expansion, command substitution, arithmetic expansion, and quote removal (see EXPANSION below).   If  the  variable  has  its integer  attribute  set,  then value is evaluated as an arithmetic expression even if the $((...)) expansion is not used (see Arithmetic Expansion below).  **Word splitting is not performed**, with the exception of "$@" as explained  below  under  Special Parameters.